packages/app/contexts/TokenListContext.tsx

```diff
@@ -163,23 +166,45 @@ export const TokenListProvider = ({ children }: PropsWithChildren) => {
             mergedTokenlistTokens,
             result.value.tokens
           );
+          const tokenUniqueAddresses: string[] = [];
+          const tokenListNoDuplicates: TokenFromTokenlist[] = [];
+
+          const tokenList = () => {
+            mergedTokenlistTokens.forEach((token) => {
+              if (!tokenUniqueAddresses.includes(token.address)) {
+                tokenUniqueAddresses.push(token.address);
+                tokenListNoDuplicates.push(token);
+              }
```

Comment:
We can already filter inside this condition, right? That will make this code faster since we aren't looping twice through the same array

Reply:
We're iterating over `mergedTokenlistTokens` here, we're not iterating twice the same array.

BUT I'm 100% agree that we add a small boolean check to know if we should push a value inside the array instead of filtering it afterward